Task.java

```diff
@@ -0,0 +1,69 @@
+public abstract class Task {
+    private String name;
+    private TaskStatus status;
+    private long schedule;
+    private Set<Task> dependsOn;
+
+    public abstract void runTask(EntityManager entityManager);
+
+    protected void finish(){
+        status = TaskStatus.Finished;
+        TaskManager.getInstance().taskFinished(this);
```

Comment:
Don't use dependencies that way, it's a caller's responsibility to call `taskManager.taskFinished(task)`
Inside taskFinished method you can call `task.finish()` to change task's status to Finished.

Reply:
But how to change the code leaving it working?